skimage/restoration/_nl_means_denoising.pyx

```diff
@@ -0,0 +1,565 @@
+                if t_col == 0 and (t_pln is not 0 or t_row is not 0):
+                    alpha = 0.5
+                else:
+                    alpha = 1.
+                integral = np.zeros_like(padded)
+                for x_pln in range(x_pln_integral_min, x_pln_integral_max):
+                    for x_row in range(x_row_integral_min, x_row_integral_max):
+                        for x_col in range(x_col_integral_min,
+                                           x_col_integral_max):
+                            integral[x_pln, x_row, x_col] = \
+                                ((padded[x_pln, x_row, x_col] -
+                                padded[x_pln + t_pln, x_row + t_row,
+                                       x_col + t_col])**2 +
+                                integral[x_pln - 1, x_row, x_col] +
+                                integral[x_pln, x_row - 1, x_col] +
+                                integral[x_pln, x_row, x_col - 1] +
+                                integral[x_pln - 1, x_row - 1, x_col - 1]
+                                - integral[x_pln - 1, x_row - 1, x_col]
+                                - integral[x_pln, x_row - 1, x_col - 1]
+                                - integral[x_pln - 1, x_row, x_col - 1])
+                for x_pln in range(x_pln_dist_min, x_pln_dist_max):
+                    for x_row in range(x_row_dist_min, x_row_dist_max):
+                        for x_col in range(x_col_dist_min, x_col_dist_max):
+                            distance = _integral_to_distance_3d(integral,
+                                        x_pln, x_row, x_col, offset,
+                                        s_cube_h_square)
+                            # exp of large negative numbers is close to zero
+                            if distance > 5.:
+                                continue
+                            weight = alpha * exp(- distance)
+                            weights[x_pln, x_row, x_col] += weight
+                            weights[x_pln + t_pln, x_row + t_row,
+                                                   x_col + t_col] += weight
+                            result[x_pln, x_row, x_col] += weight * \
+                                    padded[x_pln + t_pln, x_row + t_row,
+                                                          x_col + t_col]
+                            result[x_pln + t_pln, x_row + t_row,
+                                                  x_col + t_col] += weight * \
+                                                  padded[x_pln, x_row, x_col]
+    for x_pln in range(offset, n_pln - offset):
+        for x_row in range(offset, n_row - offset):
+            for x_col in range(offset, n_col - offset):
+                # I think there is no risk of division by zero
+                # except in padded zone
+                result[x_pln, x_row, x_col] /= weights[x_pln, x_row, x_col]
+    return result[pad_size: - pad_size, pad_size: - pad_size,
```

Comment:
Please be consistent with unary `-` (no space between `-` and operand, ie, `-pad_size`). Also, PEP8 prescribes either space on either side of `:`, or no space. Always the same. ie: `pad_size:(-pad_size)` or `pad_size : -pad_size`.